clients/spark/core/src/test/scala/io/treeverse/clients/StorageUtilsSpec.scala

```diff
@@ -0,0 +1,145 @@
+package io.treeverse.clients
+
+import com.amazonaws.auth.{
+  AWSCredentialsProvider,
+  AWSStaticCredentialsProvider,
+  BasicAWSCredentials
+}
+import com.amazonaws.services.s3.model.Region
+import com.amazonaws.services.s3.{AmazonS3, AmazonS3ClientBuilder}
+import com.amazonaws.thirdparty.apache.http.HttpStatus
+import com.amazonaws.{ClientConfiguration, HttpMethod, Protocol, SdkClientException}
+import okhttp3.HttpUrl
+import okhttp3.mockwebserver.{MockResponse, MockWebServer, RecordedRequest}
+import org.scalatest.BeforeAndAfter
+import org.scalatest.funspec.AnyFunSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatestplus.mockito.MockitoSugar
+
+class StorageUtilsSpec extends AnyFunSpec with BeforeAndAfter with MockitoSugar with Matchers {
+  private val credentialsProvider: AWSCredentialsProvider = new AWSStaticCredentialsProvider(
+    new BasicAWSCredentials("ACCESS_KEY", "SECRET_KEY")
+  )
+
+  private var awsS3ClientBuilder: AmazonS3ClientBuilder = null
+  private var server: MockWebServer = null
+  private var clientConfiguration: ClientConfiguration = null
+
+  private val US_STANDARD = "US"
+  private val US_WEST_2 = "us-west-2"
+  private val BUCKET_NAME = "bucket"
+
+  before {
+    server = new MockWebServer
+    server.start()
+    awsS3ClientBuilder = AmazonS3ClientBuilder.standard().withPathStyleAccessEnabled(true)
+    val baseUrl = server.url("/")
+    clientConfiguration = generateS3ClientConfigurations(baseUrl)
+  }
```

Comment:
I don't understand this function (AFAIU `before` is just a function that takes a function written in a weird way) in conjunction with all the `var`s above.  If we had this block:
```scala
  private val awsS3ClientBuilder  = AmazonS3ClientBuilder.standard().withPathStyleAccessEnabled(true)
  private val server = new MockWebServer
  private val clientConfiguration = {
    val baseUrl = server.url("/)
    generateS3ClientConfigurations(baseUrl)
  }
```
then we could have here just
```suggestion
  before {
    server.start()
  }
```

I believe that would be clearer, and _add_ immutability.

Reply:
`server.start` will fail if not re-initializing the mock server... It's the package's fault.
I'll reposition the others.
Thanks!